mungegithub/mungers/flake-manager.go

```diff
@@ -270,5 +278,55 @@ func (p *brokenJobSource) Body(newIssue bool) string {
 
 // Labels implements IssueSource
 func (p *brokenJobSource) Labels() []string {
-	return []string{"kind/flake", "team/test-infra"}
+	return []string{"kind/flake", "team/test-infra", sync.PriorityP2.String()}
+}
+
+// Priority implements IssueSource
+func (p *brokenJobSource) Priority(obj *github.MungeObject) (sync.Priority, error) {
+	// Different IssueSource's Priority calculation may differ
+	return autoPrioritize(obj)
+}
+
+// autoPrioritize takes a flaky test issue and prioritize it based on the number of flakes
+func autoPrioritize(obj *github.MungeObject) (sync.Priority, error) {
+	comments, err := obj.ListComments()
+	if err != nil {
+		return sync.PriorityP2, fmt.Errorf("Failed to list comment of issue")
+	}
+	occurence := []time.Time{*obj.Issue.CreatedAt}
+	lastMonth := time.Now().Add(-1 * 30 * 24 * time.Hour)
+	lastWeek := time.Now().Add(-1 * 7 * 24 * time.Hour)
+	// number of flakes happened in this month
+	monthCount := 0
+	// number of flakes happened in this week
+	weekCount := 0
+
+	for _, c := range comments {
+		// "Failed:" is a special string contained in flake issue filed by flake-manager
+		// Please make sure it matches the body generated by IssueSource.Body()
+		if !sync.RobotUser.Has(*c.User.Login) || !strings.Contains(*c.Body, "Failed:") {
+			continue
+		}
+		occurence = append(occurence, *c.CreatedAt)
+	}
+
+	for _, o := range occurence {
+		if lastMonth.Before(o) {
+			monthCount += 1
+		}
+		if lastWeek.Before(o) {
+			weekCount += 1
+		}
+	}
+
+	// P2: By default
+	// P1: Flake happens more than once in last month.
+	// P0: Flake happens more than once in last week.
+	p := sync.PriorityP2
+	if weekCount >= 2 {
```

Comment:
I think we probably want the bot to increase the priority even if a human had previously set it lower. Since an additional occurrence could mean it is now more important.